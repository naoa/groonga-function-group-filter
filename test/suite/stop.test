plugin_register functions/group_filter
table_create applicants TABLE_HASH_KEY ShortText
table_create ipcs TABLE_HASH_KEY ShortText
table_create docs TABLE_HASH_KEY ShortText
column_create docs applicants COLUMN_VECTOR applicants
column_create docs ipcs COLUMN_VECTOR ShortText
load --table docs
[
{"_key": "JP2001213456", "applicants": ["パナソニック", "三洋オプテックデザイン"], "ipcs": ["G06F", "G04A"]},
{"_key": "JP2001213457", "applicants": ["パナソニック"], "ipcs": ["G04A", "G06F"]},
{"_key": "JP2001213458", "applicants": ["ソニー"], "ipcs": ["G06F", "G04B"]},
{"_key": "JP2001213459", "applicants": ["パナソニック", "三洋電機"], "ipcs": ["G06N"]},
{"_key": "JP2001213452", "applicants": ["パナソニック", "ソニー"], "ipcs": ["G06N"]}
]
column_create applicants applicants COLUMN_INDEX docs applicants
column_create ipcs ipcs COLUMN_INDEX docs ipcs

column_create applicants synonym COLUMN_SCALAR ShortText
column_create applicants is_filter_column COLUMN_SCALAR Bool

load --table applicants
[
{"_key": "三洋電機", "synonym": "パナソニック"},
{"_key": "ソニー", "is_filter_column": true}
]

select docs \
  --filter '_id >= 1 && group_filter(applicants, 3, "synonym || _key","is_filter_column")' \
  --output_columns _id,_score,_key,applicants,#group_applicants \
  --drilldowns[sum].keys #group_applicants
