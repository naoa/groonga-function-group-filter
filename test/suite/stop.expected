plugin_register functions/group_filter
[[0,0.0,0.0],true]
table_create applicants TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create ipcs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create docs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create docs applicants COLUMN_VECTOR applicants
[[0,0.0,0.0],true]
column_create docs ipcs COLUMN_VECTOR ShortText
[[0,0.0,0.0],true]
load --table docs
[
{"_key": "JP2001213456", "applicants": ["パナソニック", "三洋オプテックデザイン"], "ipcs": ["G06F", "G04A"]},
{"_key": "JP2001213457", "applicants": ["パナソニック"], "ipcs": ["G04A", "G06F"]},
{"_key": "JP2001213458", "applicants": ["ソニー"], "ipcs": ["G06F", "G04B"]},
{"_key": "JP2001213459", "applicants": ["パナソニック", "三洋電機"], "ipcs": ["G06N"]},
{"_key": "JP2001213452", "applicants": ["パナソニック", "ソニー"], "ipcs": ["G06N"]}
]
[[0,0.0,0.0],5]
column_create applicants applicants COLUMN_INDEX docs applicants
[[0,0.0,0.0],true]
column_create ipcs ipcs COLUMN_INDEX docs ipcs
[[0,0.0,0.0],true]
column_create applicants synonym COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create applicants is_filter_column COLUMN_SCALAR Bool
[[0,0.0,0.0],true]
load --table applicants
[
{"_key": "三洋電機", "synonym": "パナソニック"},
{"_key": "ソニー", "is_filter_column": true}
]
[[0,0.0,0.0],2]
select docs   --filter '_id >= 1 && group_filter(applicants, 3, "synonym || _key","is_filter_column")'   --output_columns _id,_score,_key,applicants,#group_applicants   --drilldowns[sum].keys #group_applicants
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        4
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_score",
          "Int32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "applicants",
          "applicants"
        ],
        [
          "#group_applicants",
          "applicants"
        ]
      ],
      [
        1,
        3,
        "JP2001213456",
        [
          "パナソニック",
          "三洋オプテックデザイン"
        ],
        [
          "パナソニック",
          "三洋オプテックデザイン"
        ]
      ],
      [
        2,
        2,
        "JP2001213457",
        [
          "パナソニック"
        ],
        [
          "パナソニック"
        ]
      ],
      [
        4,
        3,
        "JP2001213459",
        [
          "パナソニック",
          "三洋電機"
        ],
        [
          "パナソニック"
        ]
      ],
      [
        5,
        2,
        "JP2001213452",
        [
          "パナソニック",
          "ソニー"
        ],
        [
          "パナソニック"
        ]
      ]
    ],
    {
      "sum": [
        [
          2
        ],
        [
          [
            "_key",
            "ShortText"
          ],
          [
            "_nsubrecs",
            "Int32"
          ]
        ],
        [
          "パナソニック",
          4
        ],
        [
          "三洋オプテックデザイン",
          1
        ]
      ]
    }
  ]
]
